src/components/map/GpxService.js

```diff
@@ -0,0 +1,74 @@
+goog.provide('ga_gpx_service');
+
+goog.require('ga_styles_service');
+
+(function() {
+
+  var module = angular.module('ga_gpx_service', [
+    'ga_styles_service'
+  ]);
+
+  /**
+   * Service used by the gaVector service to have GPX layers configuration.
+   */
+  module.provider('gaGpx', function() {
+
+    this.$get = function($q, gaStyleFactory) {
+
+      var format; // GPX parser
+      var style; // GPX default style
+
+      var Gpx = function() {
+
+        this.getType = function() {
+          return 'GPX';
+        };
+
+        // Returns the unique GPX format used by the application
+        this.getFormat = function() {
+          if (!format) {
+            format = new ol.format.GPX({
+              readExtensions: false
+            });
+          }
+          return format;
+        };
+
+        // Returns the default style for GPX feature
+        this.getStyle = function() {
+          if (!style) {
+            style = gaStyleFactory.getStyle('kml');
+          }
+          return style;
+        };
+
+        // Sanitize the content of a GPX file.
+        this.sanitize = function(gpx) {
+          return gpx;
+        };
+
+        // Sanitize a feature.
+        this.sanitizeFeature = function(feature) {
+          feature.setStyle([this.getStyle()]);
+          return feature;
+        };
+
+        // Get name of the layer.
+        this.getName = function(gpx) {
+          var name = (gpx.match(/<metadata>\s*<name>(.*?)<\/name>/) || [])[1];
```

Comment:
It can contain a name, it is optional in the spec. Maybe should we have a default name `GPX track` and move that in the if condition.

Reply:
I already put a generic default value in the gaVectorService https://github.com/geoadmin/mf-geoadmin3/pull/3962/files#diff-9a44cf53146359d60d3bf77fa1a4d822R140